src/tools/components/toolbay_form.tsx

```diff
@@ -0,0 +1,200 @@
+                                            <BlurableInput
+                                                value={String(x)}
+                                                onCommit={update}
+                                                />
+                                        </td>
+                                        <td>
+                                            <BlurableInput
+                                                value={String(y)}
+                                                onCommit={update}
+                                                />
+                                        </td>
+                                        <td>
+                                            <BlurableInput
+                                                value={String(z)}
+                                                onCommit={update}
+                                                />
+                                        </td>
+                                        <td>
+                                            <div className="select-wrapper">
+                                                <select onChange={update}>
+                                                    {tools.map(tool => {
+                                                        return <option key={
+                                                            tool.id
+                                                        }>
+                                                            {tool.name}
+                                                        </option>;
+                                                    })}
+                                                </select>
+                                            </div>
+                                        </td>
+                                        <td>
+                                            <button
+                                                className={`button-like 
+                                                    widget-control red`}
+                                                onClick={() => {
+                                                    dispatch(
+                                                        destroySlot(slot_id)
+                                                    );
+                                                } }>
+                                                X
+                                            </button>
+                                        </td>
+                                    </tr>;
+                                })}
+                                <tr>
+                                    <td>
+                                        {slotNum + 1}
+                                    </td>
+                                    <td>
+                                        <BlurableInput
+                                            value={String(this.state.x)}
```

Comment:
I think I may have left some of these behind carelessly in the past. Sorry for any confusion. Please clean them up as you find them.